### PR TITLE
[gRNAet7B] Changes from neo4j/apoc#250

### PR DESCRIPTION
**Pull request by vga91**
Changes from https://github.com/neo4j/apoc/pull/250
- Added tests 
- Simplified implementation by adding `TriggerInfo.java`

Fixed adoc bug by adding `[]` at the end of the includes

### CODE REVIEW

File: core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java [hunk 1]
```diff
@@ -45,89 +34,84 @@ public static void checkEnabled() {
...
             node.setProperty(SystemPropertyKeys.params.name(), Util.toJson(params));
             node.setProperty(SystemPropertyKeys.paused.name(), false);
 
             setLastUpdate(databaseName, tx);
-            return null;
         });
 
-        return previous;
+        return previous.get();
     }
```
Comment (AzuObs) [inline core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java:57]: Hi Giuseppe, I didn't notice this in the dev PR but could you explain why you think you need an atomic reference here? 

AtomicReference is a concurrent programming construct, but this function is executed in a single thread and has no mutable state which is accessible from another thread. So the way I see it, it shouldn't be needed.

Comment (AzuObs) [reply core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java:57]: It's not really doing any harm, but it's just confusing to see here and makes the reader question what is happening and why it's needed, which doesn't help with maintainability. So if there is no use for it, could we please remove it from here and dev too?

**Review by AzuObs — CHANGES_REQUESTED** [review]
Great backport 😸 One small nitpick.

File: core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java (hunk 1 above)
Comment (vga91) [reply core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java:56]: Hi @AzuObs, I used an `AtomicReference` just to get rid of the `Variable used in lambda expression should be final or effectively final` compilation's error within the lambda.

I just changed them to a one element final array

Comment (AzuObs) [reply core/src/main/java/apoc/trigger/TriggerHandlerNewProcedures.java:56]: Apologies, now that you mention, I remember us having a conversation about this before. Thanks for making the change.

**Review by AzuObs — APPROVED** [review]

**Pull request merged**